sources/corepackUtils.ts

```diff
@@ -400,12 +400,16 @@ export async function runVersion(locator: Locator, installSpec: InstallSpec & {s
   if (!binPath)
     throw new Error(`Assertion failed: Unable to locate path for bin '${binName}'`);
 
-  // Node.js segfaults when using npm@>=9.7.0 and v8-compile-cache
-  // $ docker run -it node:20.3.0-slim corepack npm@9.7.1 --version
-  // [SIGSEGV]
-  if (locator.name !== `npm` || semverLt(locator.reference, `9.7.0`))
-    // @ts-expect-error - No types
-    await import(`v8-compile-cache`);
+  // @ts-expect-error - Missing types
+  if (!Module.enableCompileCache) {
+    // Node.js segfaults when using npm@>=9.7.0 and v8-compile-cache
+    // $ docker run -it node:20.3.0-slim corepack npm@9.7.1 --version
+    // [SIGSEGV]
+    if (locator.name !== `npm` || semverLt(locator.reference, `9.7.0`)) {
```

Comment:
nit: we can minimize the diff, to avoid cluttering git blame
```suggestion
  // Node.js segfaults when using npm@>=9.7.0 and v8-compile-cache
  // $ docker run -it node:20.3.0-slim corepack npm@9.7.1 --version
  // [SIGSEGV]
  if (locator.name !== `npm` || semverLt(locator.reference, `9.7.0`)) {
    // @ts-expect-error - Missing types
    if (!Module.enableCompileCache) {
```

Reply:
The suggested change would exclude npm from the code cache which we don't need to do.

Reply:
I haven't benchmarked it but this would cause it to do work it doesn't have to (semverLt)

Reply:
Only for npm though

Reply:
It would always hit npm as it has to check the version.

Reply:
Yeah that's what I'm saying, and if native support is indeed disabled – otherwise it would have to do the check anyway